numpy/lib/polynomial.py

```diff
@@ -489,16 +489,19 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
         default) just the coefficients are returned, when True diagnostic
         information from the singular value decomposition is also returned.
     w : array_like, shape (M,), optional
-        Weights to apply to the y-coordinates of the sample points. For
-        gaussian uncertainties, use 1/sigma (not 1/sigma**2).
+        Weights. If not None, the weight to apply to the unsquared residual
+        at each point ``(x[i], y[i])``. Ideally the weights are chosen so
```

Comment:
This sentence doesn't read quite right; I tried to patch it up below.

```suggestion
        Weights. If not None, the weight ``w[i]`` applies to the unsquared residual
        ``y[i] - y_hat[i]`` at ``x[i]``. Ideally the weights are chosen so
```

Not sure what the correct expression for residual shouldbe.

Reply:
It was meant to be a noun phrase, rather than a sentence.  I think all the rest of the docs start with a noun phrase, rather than a sentence.  Maybe that's more clear if I remove the first "Weights." noun phrase.

Reply:
I think what you suggested also reads well.  Leaving "Weights." and then making the next thing a real grammatical sentence.  So I accepted that and copied that over to the other locations.  Thanks!

Reply:
Ah!  Now that you say it, I can read it that way.